CanvasPlusPlayground/Features/Pinned Items/PinnedItemCard.swift

```diff
@@ -176,4 +176,17 @@ extension View {
                     .fill(.secondary.opacity(selected ? 0.30 : 0.15))
             }
     }
+
+    func pinnedItemBadge(isVisible: Bool) -> some View {
+        Group {
+            if isVisible {
+                self
+                Image(systemName: "pin.fill")
+                    .font(.caption2)
+                    .foregroundStyle(.orange)
+            } else {
+                self
+            }
+        }
+    }
```

Comment:
Have the modifier itself enclose the view in an HStack(spacing: 4) so you don't need to do it in each place.


```suggestion
    func pinnedItemBadge(isVisible: Bool) -> some View {
        Group {
            if isVisible {
                HStack(spacing: 4) {
                    self
                    Image(systemName: "pin.fill")
                       .font(.caption2)
                       .foregroundStyle(.orange)
                }
            } else {
                self
            }
        }
    }
```

Reply:
You also don't need the `Group`